Profile almost done (missing edit)

bm-vs opened the pull request.

bm-vs committed:
user routes fixed

bm-vs committed:
profile page started

bm-vs committed:
user profile layout done

bm-vs committed:
merge

bm-vs committed:
friends/blocking done

bm-vs committed:
confirmation modal for block/friend actions

bm-vs committed:
donation working

bm-vs committed:
loan working

bm-vs committed:
donation/loan request

Evenilink commented:
You can still see the "add friend" and "block user" when you're already logged in as yourself.

diogohmcruz reviewed (changes requested):
Ver @Evenilink comment

diogohmcruz commented on client/src/components/User.js:
```diff
@@ -1,56 +1,797 @@
 
-import { Container, Header, Segment } from 'semantic-ui-react';
+import { Container, Header, Divider, Button, Image, Icon, Rating, Modal, Input } from 'semantic-ui-react';
+import { withCookies, Cookies } from 'react-cookie';
```
Há um warning que diz que não estás a usar o Cookies.
Ou te esqueces te de usar onde querias, ou é para tirar.

diogohmcruz committed:
Merge branch 'develop' into feature/api/profile

bm-vs committed:
fixed eslint warning and added authentication to donation/loan request

bm-vs committed:
prep merge

bm-vs committed:
resolved merge

bm-vs committed:
edit profile working

diogohmcruz commented:
@BSantos22 broken atm.

bm-vs committed:
menu for unknown user done

bm-vs committed:
menu for friend started

bm-vs committed:
finished profile buttons

bm-vs committed:
merge conflicts fixed

diogohmcruz committed:
Cleanup

- Clean code smells;
- Tirei os localhosts dos url;
- Meti os containers dos divider fluid;
- troquei alguns if para ?, torna mais legivel.
